### DISCUSSION TITLE
Unrecognized shell error on Mac

### DISCUSSION THREAD

**Discussion by gudstra** [Taitava/obsidian-shellcommands#222] [Debugging and testing]
I see this error a lot. One way to replicate the issue is to simplt open the settings in Obsidian.  Things still seem to work but it is very annoying.

Error:
EscapeValue(): Unrecognised shell: sh

![CleanShot 2022-05-13 at 15 35 42@2x](https://user-images.githubusercontent.com/104004300/168386142-8e4ca263-e2c3-43c7-bf4d-3153677a12bd.png)

**Comment by Taitava** [comment]
The main problem here is that the _Shell commands_ plugin cannot recognise what kind of special characters escaping method to use when the shell is `sh`. `sh` is a short name that can be used by many different shells. What is your default shell in macOS?

On macOS and Linux, the plugin supports `bash`, `dash`, and `zsh`. If you have one of those installed, the issue is that SC does not recorgnise it, because it sees just `sh`. Assuming you have one of those shells installed, you can [change the default shell option in SC's settings](https://publish.obsidian.md/shellcommands/Environments/Shells#How+to+know+which+shell+is+used) to solve this issue. Select something else than _Use system default_.

> One way to replicate the issue is to simplt open the settings in Obsidian. Things still seem to work but it is very annoying.

In settings, this error message probably comes up when shell command preview texts are being displayed. It might cause some things not to work correctly in the settings, as it stops JavaScript code interpreting (maybe I should improve this behavior). When you actually try to execute shell commands, this error will prevent the execution, at least for shell commands that use `{{variables}}`.
